Makefile

```diff
@@ -15,7 +15,11 @@ ifeq ($(MONO),)
   MONO := /usr/bin/mono
 endif
 
-MCS := $(shell which dmcs)
+DMCS := $(shell which dmcs)
+MCS := $(shell which mcs)
+ifneq ($(DMCS),)
+  MCS := $(DMCS)
+endif
```

Comment:
Does this part need to land on something other than master? Should it go on release-5.1 or release-5.2 if it will cause problems for people trying to use mono?

Reply:
This is the part that probably needs careful review from someone with Mono knowledge. AFAICT, it should be safe. It certainly works for me, but I only built the Go bindings.

Reply:
I see. On your system, I assume you have mcs installed but not dmcs? I want to say that the correct thing is to have us switch to using mcs if available rather than dmcs, but I will say that it is probably better if we did this at another time.

Reply:
Correct; my change was conservative by only falling back to mcs if dmcs is not available, following the principle of least surprise.